index.js

```diff
@@ -2,268 +2,299 @@
+    silent: !args.loggerlevel,
+    level: args.loggerlevel,
+    format: winston.format.combine(
+       winston.format.simple(),
+       winston.format.splat(),
+    ),
+    transports: [
+      new winston.transports.Console(),
+    ],
+  })
+ 
+  LOGGER.info('Starting...');
+
+  LOGGER.debug(`Initial configuration. region="${args.region}" \
+add="${args.add}" \
+statsinterval="${args.statsinterval}" \
+json="${args.json}" \
+loggerlevel="${args.loggerlevel}" \
+events="${args.events}" \
+logs="${args.logs}" \
+secure="${args.secure}" \
+stats="${args.stats}" \
+port="${args.port}" \
+server="${args.server}"`)
+
+  if (!(args.logs || args.stats || args.events)) {
+    throw new Error('You need to enable either logs, stats or events.');
+  }
+
+  if (args.token) {
+    args.logstoken = args.logstoken || args.token;
+    args.statstoken = args.statstoken || args.token;
+    args.eventstoken = args.eventstoken || args.token;
+  }
 
-  if (argv.help || !(argv.token || argv.logstoken || argv.statstoken || argv.eventstoken) || !(argv.region)) {
-    console.log('Usage: r7insight_docker [-l LOGSTOKEN] [-k STATSTOKEN] [-e EVENTSTOKEN]\n' +
-                '                         [-t TOKEN] [--no-secure] [--json]\n' +
-                '                         [-r REGION]\n' +
-                '                         [--no-newline] [--no-stats] [--no-logs] [--no-dockerEvents]\n' +
-                '                         [-i STATSINTERVAL] [-a KEY=VALUE]\n' +
-                '                         [--matchByImage REGEX] [--matchByName REGEX]\n' +
-                '                         [--skipByImage REGEX] [--skipByName REGEX]\n' +
-                '                         [--server HOSTNAME] [--port PORT]\n' +
-                '                         [--debug]\n' +
-                '                         [--help]');
-
-    process.exit(1);
+  if (args.logs && !args.logstoken) {
+    throw new Error('Logs enabled but no log token!');
+  } else if (args.stats && !args.statstoken) {
+    throw new Error('Stats enabled but no stats token!');
+  } else if (args.events && !args.eventstoken) {
+    throw new Error('Events enabled but no events token!');
   }
 
   const getPort = () => {
-    if (argv.port == undefined) {
-      if (argv.secure) {
+    if (args.port == undefined) {
+      if (args.secure) {
         return 443;
       }
-
       return 80;
     }
 
-    // TODO: support service names
+    // TODO (@rjacobs-r7): support service names
 
-    return parseInt(argv.port);
+    return parseInt(args.port);
   };
-  argv.port = getPort();
-  if (isNaN(argv.port)) {
-    console.log('port must be a number');
-    process.exit(1);
+
+  args.port = getPort();
+  LOGGER.info(`Using port ${args.port}`);
+
+  if (isNaN(args.port)) {
+    throw new Error(`Port must be a number`);
   }
 
-  if (argv.add && !Array.isArray(argv.add)) {
-    argv.add = [argv.add];
+  LOGGER.info('Processing --add flag into valid array');
```

Comment:
```suggestion
  LOGGER.info('Processing --add flag');
```
underlying data type doesn't matter to the user, but also it's not an array :P